src/policy/policy.cpp

```diff
@@ -37,7 +37,13 @@ CAmount GetDustThreshold(const CTxOut& txout, const CFeeRate& dustRelayFeeIn)
     if (txout.scriptPubKey.IsWitnessProgram(witnessversion, witnessprogram)) {
         // sum the sizes of the parts of a transaction input
         // with 75% segwit discount applied to the script size.
-        nSize += (32 + 4 + 1 + (107 / WITNESS_SCALE_FACTOR) + 4);
+        if (witnessversion == 0) {
+            // P2WPKH spend, a compressed public key + a DER-encoded ECDSA signature
+            nSize += (32 + 4 + 1 + (107 / WITNESS_SCALE_FACTOR) + 4);
+        } else {
+            // Taproot key spend, a single BIP341 signature
+            nSize += (32 + 4 + 1 + (66 / WITNESS_SCALE_FACTOR) + 4);
```

Comment:
More of a meta-comment:

If `GetDustThreshold` attempts to calculate the cost at which it's economical to spend an output, why is `nSize` initialized to `GetSerializedSize(txout)`? When spending the dust output, the size of the txout isn't in the spending tx unless the spending tx also spends to the same output type. This heuristic/initialization doesn't make sense to me because if we take P2WSH as an example (w/ dust limit of 330), `330 - fees < 330` it is impossible to spend to the same output type unless other inputs are added. It seems to me that if witness version non-zero is being changed, this could be a place to fix or clarify this calculation for witness version non-zero.